src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -0,0 +1,518 @@
+   *  This means that references to a private type will lead to typing failures
+   *  on the code when it is inlined. Less than ideal, but hard to do better (see below).
+   *
+   *  @return If there are accessors generated, a thicket consisting of the rewritten `tree`
+   *          and all accessors, otherwise the original tree.
+   */
+  private def makeInlineable(tree: Tree)(implicit ctx: Context) = {
+
+    /** A tree map which inserts accessors for all non-public term members accessed
+     *  from inlined code. Accesors are collected in the `accessors` buffer.
+     */
+    object addAccessors extends TreeMap {
+      val inlineMethod = ctx.owner
+      val accessors = new mutable.ListBuffer[MemberDef]
+
+      /** A definition needs an accessor if it is private, protected, or qualified private */
+      def needsAccessor(sym: Symbol)(implicit ctx: Context) =
+        sym.is(AccessFlags) || sym.privateWithin.exists
+
+      /** The name of the next accessor to be generated */
+      def accessorName(implicit ctx: Context) =
+        ctx.freshNames.newName(inlineMethod.name.asTermName.inlineAccessorName.toString)
+
+      /** A fresh accessor symbol.
+       *
+       *  @param tree          The tree representing the original access to the non-public member
+       *  @param accessorInfo  The type of the accessor
+       */
+      def accessorSymbol(tree: Tree, accessorInfo: Type)(implicit ctx: Context): Symbol =
+        ctx.newSymbol(
+          owner = inlineMethod.owner,
+          name = if (tree.isTerm) accessorName.toTermName else accessorName.toTypeName,
+          flags = if (tree.isTerm) Synthetic | Method else Synthetic,
+          info = accessorInfo,
+          coord = tree.pos).entered
+
+      /** Add an accessor to a non-public method and replace the original access with a
+       *  call to the accessor.
+       *
+       *  @param tree         The original access to the non-public symbol
+       *  @param refPart      The part that refers to the method or field of the original access
+       *  @param targs        All type arguments passed in the access, if any
+       *  @param argss        All value arguments passed in the access, if any
+       *  @param accessedType The type of the accessed method or field, as seen from the access site.
+       *  @param rhs          A function that builds the right-hand side of the accessor,
+       *                      given a reference to the accessed symbol and any type and
+       *                      value arguments the need to be integrated.
+       *  @return The call to the accessor method that replaces the original access.
+       */
+      def addAccessor(tree: Tree, refPart: Tree, targs: List[Tree], argss: List[List[Tree]],
+                      accessedType: Type, rhs: (Tree, List[Type], List[List[Tree]]) => Tree)(implicit ctx: Context): Tree = {
+        val (accessorDef, accessorRef) =
+          if (refPart.symbol.isStatic) {
```

Comment:
Couldn't this be extended to:
```scala
if (refPart.symbol.isStatic || refPart.symbol.isContainedIn(inlineMethod.owner) {
  ...
}
```
This way when we have:
```scala
    private val x: Int
    inline def get1 = x
```
we'll get:
```scala
def $inlineAccessor$get1$1: T = C.this.x
```
instead of:
```scala
def $inlineAccessor$get1$1(x$0: Test.C[T]): T = x$0.x
```